packages/shopify-codemod/transforms/split-if-assignments.js

```diff
@@ -1,13 +1,32 @@
 export default function splitIfAssignments({source}, {jscodeshift: j}, {printOptions = {}}) {
-  return j(source)
-    .find(j.IfStatement, {
-      test: {
-        type: j.AssignmentExpression.name,
-      },
-    }).forEach((path) => {
-      const assignment = path.get('test');
-      path.insertBefore(j.expressionStatement(assignment.node));
-      assignment.replace(assignment.node.left);
-    })
-    .toSource(printOptions);
+
+  const sourceAST = j(source);
+
+  // Handle else-ifs in reverse order (otherwise initial transforms orphan later paths).
+  sourceAST.find(j.IfStatement, {
+    alternate: {
+      type: j.IfStatement.name,
+      test: j.AssignmentExpression.check,
+    },
+  }).paths()
+    .reverse()
+    .forEach(({node}) => {
+      const elseIf = node.alternate;
+      const test = elseIf.test;
+      node.alternate = j.blockStatement([
```

Comment:
I am a little rusty on this, but I seem to remember preferring using the path and using `replace()` instead of directly mutating the node.